dog_lover/src/breedContext.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useContext, useState } from "react";
+import axios from "axios";
+import {
+  DogBreed,
+  ListAllResponse,
+  ImageResponse,
+  DogBreedImage,
+} from "../components/breedSelectorBox";
+
+const ClickedBreedContext = React.createContext({} as DogBreedImage[]);
+const BreedContext = React.createContext({} as DogBreed[]);
+const HeartContextUpdate = React.createContext({} as any);
+const FetchBreedsContextUpdate = React.createContext({} as any);
+const OpenContext = React.createContext({} as boolean);
+const QueryContext = React.createContext({} as any);
+const QueryContextUpdate = React.createContext({} as any);
+
+export const useVariableBreedContext = () => {
+  return useContext(BreedContext);
+};
+export const useVariableClickedBreedContext = () => {
+  return useContext(ClickedBreedContext);
+};
+export const useVariableOpenContext = () => {
+  return useContext(OpenContext);
+};
+export const useVariableQueryContext = () => {
+  return useContext(QueryContext);
+};
+export const useFunctionHeartContext = () => {
+  return useContext(HeartContextUpdate);
+};
+export const useFunctionFetchBreedsContext = () => {
+  return useContext(FetchBreedsContextUpdate);
+};
+export const useFunctionQueryContext = () => {
+  return useContext(QueryContextUpdate);
+};
+
+export const BreedProvider = ({ children }: { children: any }) => {
```

Comment:
How did you named this BreedProvider? I mean it's not a Component?

Reply:
Well, it's my first time with the context API in general, I learned all I needed to know about it through an awesome YouTuber called Web Dev Simplified, so I don't quite understand your question: is it usually impossible to change its name? To be honest, now that I see it, I think I should have named it "DogProvider", it makes much more sense :D

Reply:
My question was derived from the file name. I expected to see context related code here, and the rest in a different file like `DogProvider` or `Dogs` under the components or pages directory.

Reply:
Yes, that would make sense. I am without a doubt new with the Context API so I am not exactly familiar with the best practices around it.